datamodel/create_roles.py

```diff
@@ -0,0 +1,237 @@
+        ALTER DEFAULT PRIVILEGES IN SCHEMA {modelname}_sys GRANT ALL ON SEQUENCES TO {roles[3]};
+        """
+    cur.execute(grant_sql)
+    if ext_schema:
+        grant_sql = f"""
+        GRANT USAGE ON SCHEMA {ext_schema}  TO {roles[0]};
+        GRANT USAGE, SELECT ON ALL SEQUENCES IN SCHEMA {ext_schema}  TO {roles[0]};
+        GRANT SELECT, REFERENCES, TRIGGER ON ALL TABLES IN SCHEMA {ext_schema}  TO {roles[0]};
+        ALTER DEFAULT PRIVILEGES IN SCHEMA {ext_schema}  GRANT SELECT, REFERENCES, TRIGGER ON TABLES TO {roles[0]};
+        /* User */
+        GRANT ALL ON SCHEMA {ext_schema} TO {roles[1]};
+        GRANT ALL ON ALL TABLES IN SCHEMA {ext_schema} TO {roles[1]};
+        GRANT ALL ON ALL SEQUENCES IN SCHEMA {ext_schema} TO {roles[1]};
+         ALTER DEFAULT PRIVILEGES IN SCHEMA {ext_schema} GRANT ALL ON TABLES TO {roles[1]};
+        """
+        cur.execute(grant_sql)
+    conn.commit()
+    conn.close()
+
+
+def revoke_privileges(pg_service: str, modelname: str, roles: list, ext_schema: str = None):
+    conn = psycopg.connect(f"service={pg_service}")
+    cur = conn.cursor()
+    revoke_sql = f"""
+        REVOKE ALL ON SCHEMA {modelname}_od  FROM {roles[0]};
+        REVOKE ALL ON SCHEMA {modelname}_app FROM {roles[0]};
+        REVOKE ALL ON SCHEMA {modelname}_sys FROM {roles[0]};
+        REVOKE ALL ON SCHEMA {modelname}_vl  FROM {roles[0]};
+
+        REVOKE ALL ON ALL TABLES IN SCHEMA {modelname}_od  FROM {roles[0]};
+        REVOKE ALL ON ALL TABLES IN SCHEMA {modelname}_app FROM {roles[0]};
+        REVOKE ALL ON ALL TABLES IN SCHEMA {modelname}_sys FROM {roles[0]};
+        REVOKE ALL ON ALL TABLES IN SCHEMA {modelname}_vl  FROM {roles[0]};
+
+        ALTER DEFAULT PRIVILEGES IN SCHEMA {modelname}_od  REVOKE ALL ON TABLES  FROM {roles[0]};
+        ALTER DEFAULT PRIVILEGES IN SCHEMA {modelname}_app REVOKE ALL ON TABLES  FROM {roles[0]};
+        ALTER DEFAULT PRIVILEGES IN SCHEMA {modelname}_sys REVOKE ALL ON TABLES  FROM {roles[0]};
+        ALTER DEFAULT PRIVILEGES IN SCHEMA {modelname}_vl  REVOKE ALL ON TABLES  FROM {roles[0]};
+        """
+    cur.execute(revoke_sql)
+    if ext_schema:
+        revoke_sql = f"""
+        REVOKE ALL ON SCHEMA {ext_schema} FROM {roles[0]};
+        REVOKE ALL ON ALL TABLES IN SCHEMA {ext_schema} FROM {roles[0]};
+        ALTER DEFAULT PRIVILEGES IN SCHEMA {ext_schema} REVOKE ALL ON TABLES  FROM {roles[0]};
+        """
+        cur.execute(revoke_sql)
+    conn.commit()
+    conn.close()
+
+
+def create_roles(
+    modelname: str,
+    pg_service: str = None,
+    grant: bool = True,
+    ext_schema: str = None,
+    db_spec_roles: Optional[bool] = False,
+):
+    """
+    Creates the roles for usage in TEKSI Modules
+    :param pg_service: the PostgreSQL service
+    :param modelname: Abbreviation of the TEKSI model
+    :param grant: Boolean defining whether to grant privileges or revoke them
+    :param ext_schema: Schema name of the extension model
```

Comment:
can you be a bit more descriptive? 
is this already fixed how this should behave?
can there be more than one extension schema?